### PR TITLE
fix: message not found on quotes of your own messages [SQSERVICES-1834]

### PR DESCRIPTION
**Pull request by tlebon**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/SQSERVICES-1834" title="SQSERVICES-1834" target="_blank">BugSQSERVICES-1834</a>  "You cannot see this message." in reply to own message
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

fix an issue with quotes of your own (edited) messages not appearing

### Causes (Optional)

reference to message id is changing, using replacing_message_id. also an issue with race condition for hashed quotes.

### CODE REVIEW

File: src/script/event/EventService.ts [hunk 1]
```diff
@@ -127,18 +127,30 @@ export class EventService {
...
       }
 
       const records = (await this.storageService.getAll(StorageSchemata.OBJECT_STORE.EVENTS)) as EventRecord[];
       return records
-        .filter(record => record.id === eventId && record.conversation === conversationId)
+        .filter(
+          record =>
+            (record.id === eventId && record.conversation === conversationId) ||
+            (record.id === eventId && record.data.replacing_message_id === eventId),
+        )
```
Comment (tlebon) [inline src/script/event/EventService.ts]: not sure how important this is.

File: src/script/notification/NotificationRepository.ts [hunk 2]
```diff
@@ -259,7 +259,9 @@ export class NotificationRepository {
             const messageInfo = messageId
```
Comment (tlebon) [inline src/script/notification/NotificationRepository.ts:259]: all the edits in this file are just improving some logger [object Object] issues

File: src/script/event/preprocessor/QuotedMessageMiddleware.ts [hunk 3]
```diff
@@ -117,24 +116,10 @@ export class QuotedMessageMiddleware {
       return {...event, data: decoratedData};
     }
 
-    const quotedMessageHash = new Uint8Array(quote.quotedMessageSha256).buffer;
-
-    const isValid = await MessageHasher.validateHash(quotedMessage, quotedMessageHash);
-    let quoteData;
-
-    if (!isValid) {
```
Comment (tlebon) [inline src/script/event/preprocessor/QuotedMessageMiddleware.ts:125]: confirmed with @comawill this is fine not confirming hashes if the message is found in db

File: src/script/event/EventService.ts [hunk 4]
```diff
@@ -127,18 +127,30 @@ export class EventService {
 
     try {
       if (this.storageService.db) {
```
Comment (atomrc) [inline src/script/event/EventService.ts]: in which case do we not have a `db`? Did you find a reason why we have those 2 branches?

Comment (tlebon) [reply src/script/event/EventService.ts]: i could never get that branch to go off, not sure tbh

Comment (atomrc) [reply src/script/event/EventService.ts]: oohhh it those experiment where we have different DB engines under the hood 😠 
image

Comment (atomrc) [reply src/script/event/EventService.ts]: I guess we need to handle both scenarios 😬

File: src/script/event/EventService.ts [hunk 5]
```diff
@@ -127,18 +127,30 @@ export class EventService {
...
           .first();
+        if (!entry) {
+          entry = await this.storageService.db
+            .table(StorageSchemata.OBJECT_STORE.EVENTS)
+            .where('conversation')
+            .equals(conversationId)
+            .filter(item => item.data.replacing_message_id === eventId)
+            .first();
+        }
         return entry;
```
Comment (atomrc) [inline src/script/event/EventService.ts]:
```suggestion
        if (entry) {
          return entry;
        }
        return await this.storageService.db
          .table(StorageSchemata.OBJECT_STORE.EVENTS)
          .where('conversation')
          .equals(conversationId)
          .filter(item => item.data.replacing_message_id === eventId)
          .first();
```

Comment (atomrc) [reply src/script/event/EventService.ts]: This way you can avoid the `let` on L130. WDYT?

File: src/script/event/EventService.ts (hunk 1 above)
Comment (atomrc) [reply src/script/event/EventService.ts]: It's for when we don't have a dexie DB under the hood (not sure if we have valid scenarios where we have a different engine... 🤔 ). But let's keep it there 👍

File: test/unit_tests/event/preprocessor/QuotedMessageMiddlewareSpec.js [hunk 6]
```diff
@@ -81,44 +81,6 @@ describe('QuotedMessageMiddleware', () => {
...
     });
 
-    it('adds an error if hashes do not match', async () => {
-      const expectedError = {
-        type: QuoteEntity.ERROR.INVALID_HASH,
-      };
-
-      const quotedMessage = {
-        data: {
-          content: 'pas salut',
```
Comment (V-Gira) [inline test/unit_tests/event/preprocessor/QuotedMessageMiddlewareSpec.js:91]: 'salut'

Really wonder who wrote this test :thinking:

Comment (atomrc) [reply test/unit_tests/event/preprocessor/QuotedMessageMiddlewareSpec.js:91]: ahaha guess who implemented those tests 😆

File: src/script/event/EventService.ts (hunk 4 above)
Comment (tlebon) [reply src/script/event/EventService.ts:129]: should we remove it?

Comment (tlebon) [reply src/script/event/EventService.ts]: okay then

File: src/script/event/EventService.ts (hunk 1 above)
Comment (atomrc) [reply src/script/event/EventService.ts]: For reference, it was introduced here with temporary guest devices https://github.com/wireapp/wire-webapp/pull/7086/files#diff-6ab223d7041cd0190d856a1ee68a04f4c60175878ff0fcc77810a789569eeff7

**Review by atomrc — APPROVED** [review]

**Pull request merged**